nasa/harmony-browse-image-generator
# Implement pre-commit hooks.

## Description

This PR begins the process of applying automated checking to all commits via pre-commit hooks. Currently it checks a small number of things:

* Trailing whitespace
* Blank lines at the end of the file.
* Valid JSON and YAML syntax.
* `ruff` linting.
* `black` code formatting

Why haven't I added `mypy`? So far all these checks have left the service code untouched. I think we should totally add both `mypy`, but just wanted to get the ball rolling here.

## Jira Issue ID

N/A - this is an IP-ish bit of fun.

## Local Test Steps

* Pull this branch.
* Follow the new instructions in the README to install `pre-commit`.
* Make some minor change, use `git add` then `git commit`. You should see the checks all run when you are trying to use `git commit`.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~`CHANGELOG.md` updated to include high level summary of PR changes.~~
* ~~`VERSION` updated if publishing a release.~~
* [x] Tests added/updated and passing. (Well, the hooks are all passing, but that's not quite the same thing)
